jc/parsers/find.py

```diff
@@ -0,0 +1,126 @@
+"""jc - JSON Convert `find` command output parser
+
+Usage (cli):
+
+    $ find | jc --find
+
+Usage (module):
+
+    import jc
+    result = jc.parse('find', find_command_output)
+
+Schema:
+
+    [
+      {
+        "path":     string,
+        "node":     string,
+        "error":    string
+      }
+    ]
+
+Examples:
+
+    $ find | jc --find -p
+    [
+        {
+          "path": "./directory"
+          "node": "filename"
+          "error": ""
+        },
+        {
+          "path": "./anotherdirectory"
+          "node": "anotherfile"
+          "error": ""
+        },
+        ...
+    ]
+
+    $ find | jc --find -p -r
+    [
+      "./templates/readme_template",
+      "./templates/manpage_template",
+      "./.github/workflows/pythonapp.yml",
+      ...
+    ]
+"""
+import jc.utils
+
+
+class info():
+    """Provides parser metadata (version, author, etc.)"""
+    version = '1.0'
+    description = '`find` command parser'
+    author = 'Solomon Leang'
+    author_email = 'solomonleang@gmail.com'
+    compatible = ['linux']
+    tags = ['command']
+
+
+__version__ = info.version
+
+
+def _process(proc_data):
+    """
+    Final processing to conform to the schema.
+
+    Parameters:
+
+        proc_data:   (Dictionary) raw structured data to process
+
+    Returns:
+
+        List of Dictionaries. Structured data to conform to the schema.
+    """
+
+    processed = []
+    for index in proc_data:
+        path, node, error = "", "", ""
+        if (index == "."):
+            node = "."
+        elif (index[:4] == "find"):
+            error = index
+        else:
+            try:
+                path, node = index.rsplit('/', maxsplit=1)
+            except ValueError:
+                pass
+        
+        proc_line = {
+            'path': path,
+            'node': node,
+            'error': error
+        }
```

Comment:
We could do something like this so error field only shows up when needed:
```
proc_line = {
    'path': path if path else None,
    'node': node if node else None
}

if error:
    proc_line.update(
        {'error': error}
    )
```

That will make the values `null` instead of blank strings if they are empty which seems cleaner to me.

Reply:
Yep! Just updated for this. I agree it does look cleaner.